zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/DataSetInfo.scala

```diff
@@ -13,3 +18,120 @@ case class DataSetInfo(name: String,
                        schema: Schema,
                        dataInterval: Interval,
                        stats: Stats)
+
+object DataSetInfo {
+
+  def parse(json: JsValue): DataSetInfo = {
+    json.validate[DataSetInfo] match {
+      case js: JsSuccess[DataSetInfo] => js.get
+      case e: JsError => throw JsonRequestException(JsError.toJson(e).toString())
+    }
+  }
+
+  def write(dataSetInfo: DataSetInfo): JsValue = Json.toJson(dataSetInfo)
+
+  implicit val intervalFormat: Format[Interval] = new Format[Interval] {
+    override def reads(json: JsValue) = {
+      val start = (json \ "start").as[DateTime]
+      val end = (json \ "end").as[DateTime]
+      JsSuccess(new Interval(start.getMillis, end.getMillis))
+    }
+
+    override def writes(interval: Interval): JsValue = {
+      val formatter = ISODateTimeFormat.dateTime()
+      JsObject(List("start" -> JsString(interval.getStart.toString(formatter)), "end" -> JsString(interval.getEnd.toString(formatter))))
+    }
+  }
+
+  //Needed for HierarchyField levels
+  implicit def tuple2Reads[A, B](implicit aReads: Reads[A], bReads: Reads[B]): Reads[Tuple2[A, B]] = Reads[Tuple2[A, B]] {
```

Comment:
It is needed to read levels:Seq[(String,String)]. I made this specific for String now.